mmn/collector.go

```diff
@@ -110,41 +113,64 @@ func hashMACAddress(mac string) string {
 func getOS() string {
 	switch runtime.GOOS {
 	case "darwin":
-		output, err := shell.NewCmd("sw_vers -productVersion").Output()
+		output, err := exec.Command("sw_vers", "-productVersion").Output()
 		if err != nil {
-			log.Error().Err(err).Msg("Failed to get OS information")
-			return ""
+			log.Err(err).Msg("Failed to get OS information")
+			return "macOS (unknown)"
 		}
-		return "MAC OS X - " + strings.TrimSpace(string(output))
-
+		return "macOS " + strings.TrimSpace(string(output))
 	case "linux":
-		output, err := shell.NewCmd("lsb_release -d").Output()
+		distro, err := parseLinuxOS()
 		if err != nil {
-			output, err = shell.NewCmd("source /etc/os-release && echo $PRETTY_NAME").Output()
-			if err != nil {
-				log.Error().Err(err).Msg("Failed to get OS information")
-				return ""
-			}
+			log.Err(err).Msg("Failed to get OS information")
+			return "linux (unknown)"
 		}
-		return strings.TrimSpace(strings.Replace(string(output), "Description:", "", 1))
-
+		return distro
 	case "windows":
-		output, err := shell.NewCmd("wmic os get Caption /value").Output()
+		output, err := exec.Command("wmic", "os", "get", "Caption", "/value").Output()
 		if err != nil {
-			log.Error().Err(err).Msg("Failed to get OS information")
-			return ""
+			log.Err(err).Msg("Failed to get OS information")
+			return "windows (unknown)"
 		}
-		return extractWindowsVersion(string(output))
+		return strings.TrimSpace(strings.TrimPrefix(string(output), "Caption="))
 	}
-
-	return ""
+	return runtime.GOOS
 }
 
-func extractWindowsVersion(output string) string {
-	var version string
+func parseLinuxOS() (string, error) {
+	output, err := exec.Command("lsb_release", "-d").Output()
+	if err == nil {
+		return strings.TrimSpace(strings.TrimPrefix(string(output), "Description:")), nil
+	}
 
-	version = strings.TrimSpace(strings.Replace(output, "Caption=", "", 1))
-	version = strings.TrimSpace(strings.Replace(version, "Microsoft", "", 1))
+	const etcOsRelease = "/etc/os-release"
+	const altOsRelease = "/usr/lib/os-release"
+	osReleaseFile, err := os.Open(etcOsRelease)
```

Comment:
Much simpler would be `source /etc/os-release && echo $PRETTY_NAME`.

Reply:
I'm not sure if it will work for all OS because for CentOS this file called `/etc/centos-release`

Reply:
Oh, looks like `/etc/os-release` exists there too.

Reply:
Description mentions that source tools is not available on some environments.

Reply:
I am about complex variable parsing here.
Regarding `centos-release`, CentOS version is not needed, here we're parsing Linux version.

Reply:
Source is unavailable in alpine. It’s not really complex